lib/ansible/plugins/test/uri.py

```diff
@@ -9,7 +9,8 @@ def is_uri(value, schemes=None):
     """ Will verify that the string passed is a valid 'URI', if given a list of valid schemes it will match those """
     try:
         x = urlparse(value)
-        isit = all([x.scheme is not None, x.path is not None, not schemes or x.scheme in schemes])
+        # urlparse returns empty string for scheme when not present, not None
+        isit = all([x.scheme, x.path is not None, not schemes or x.scheme in schemes])
```

Comment:
Should we just simplify to:
```suggestion
        isit = x.path is not None and (x.scheme in schemes or ())
```
?

Reply:
To allow any schemes by default (since passing in `schemes` is optional), maybe:
```suggestion
        isit = x.path and x.scheme and (not schemes or x.scheme in schemes)
```

Reply:
Yes, you are correct and I probably should have read the docs 😬